### PR TITLE
ESLint coverage

### PR DESCRIPTION
**Pull request by chicoxyzzy**

### CODE REVIEW

**Comment by zpao** [comment]
I totally get where you're coming from but let's leave this for now. We (FB) still use jshint internally so these actually help keep things a bit quieter there. I'd be fine dropping the jshint things from everything not in `src/` and making sure we use eslint there (we don't currently), if you want to go down that route.

**Comment by chicoxyzzy** [comment]
@zpao ok. there is [one line](https://github.com/facebook/react/pull/3123/files#diff-04c6e90faac2675aa89e2176d2eec7d8L89) that should be fixed tho

**Comment by chicoxyzzy** [comment]
@zpao now ESLint is everywhere except `src/`. That was not easy :weary:

File: grunt/config/browserify.js [hunk 1]
```diff
@@ -24,8 +23,9 @@ var LICENSE_TEMPLATE =
  * All rights reserved.\n\
  *\n\
  * This source code is licensed under the BSD-style license found in the\n\
- * LICENSE file in the root directory of this source tree. An additional grant\n\
- * of patent rights can be found in the PATENTS file in the same directory.\n\
+ * LICENSE file in the root directory of this source tree. An additional\n\
+ * grant of patent rights can be found in the PATENTS file in the same\n\
+ * directory.\n\
```
Comment (zpao) [inline grunt/config/browserify.js]: Can you revert these changes. This is intended (to match the formatting of the license as it appears in regular source files).

File: grunt/tasks/eslint.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var grunt = require('grunt');
+
+module.exports = function() {
+  var done = this.async();
+  grunt.util.spawn({
+    cmd: 'node_modules/.bin/eslint',
+    args: ['src/', 'Gruntfile.js', 'grunt/', 'main.js', 'perf/', 'test/']
```
Comment (zpao) [inline grunt/tasks/eslint.js]: Hmm, can we skip specifying things specifically if we're linting everything? We can blacklist in `.eslintignore`

File: main.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 var visitors = require('./vendor/fbtransform/visitors');
 var transform = require('jstransform').transform;
 var typesSyntax = require('jstransform/visitors/type-syntax');
-var Buffer = require('buffer').Buffer;
+var Buf = require('buffer').Buffer;
```
Comment (zpao) [inline main.js]: I actually don't like this change. What option resulted in this?

File: perf/tests/propTypes.js [hunk 4]
```diff
@@ -62,22 +68,24 @@ exports.fn = function(){
...
 
   React.render(List({
     array: [11, 12, 13, 14, 15, 16, 17, 18, 19, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
     bool: false,
     number: Math.random(),
     string: 'banana banana banana',
-    func: function() { return 'this is a function'; },
+    func: function() {
+      return 'this is a function';
+    },
```
Comment (zpao) [inline perf/tests/propTypes.js]: You use `func` above but leave this one inline. Any reason?

File: perf/tests/renderComponent-basic.js [hunk 5]
```diff
@@ -1,24 +1,22 @@
...
 
 exports.setup = function(){
-  /*global*/_rootNode = document.createElement('div');
+  _rootNode = document.createElement('div');
   document.body.appendChild(_rootNode);
 };
 exports.fn = function(){
   React.render(React.DOM.div(null, 'lol, perf testing ', this.count), _rootNode);
 };
 exports.teardown = function(){
```
Comment (zpao) [inline perf/tests/renderComponent-basic.js]: eslint didn't complain about not having a space before the `{`?

File: test/lib/jasmine-execute.js [hunk 6]
```diff
@@ -1,6 +1,8 @@
-document.write('<style> @import "../vendor/jasmine/jasmine.css?_=' + (+new Date).toString(36) + '"; </style>');
+/*eslint-disable quotes*/
```
Comment (zpao) [inline test/lib/jasmine-execute.js]: The below looks right as far as quotes go. Am I missing something?

**Comment by zpao** [comment]
Overall, herculean effort. Thanks! Just a few smaller things inline I think.

Also, it looks like something got weird with your branching (or GitHub broke). Can you rebase and whatever you need to get that sorted out.

File: main.js (hunk 3 above)
Comment (chicoxyzzy) [inline main.js]: it shadows `global.Buffer` object (eslint throws `no-undef`)

File: grunt/tasks/eslint.js (hunk 2 above)
Comment (chicoxyzzy) [inline grunt/tasks/eslint.js]: seems like file is required
http://eslint.org/docs/command-line-interface/

File: perf/tests/propTypes.js (hunk 4 above)
Comment (chicoxyzzy) [inline perf/tests/propTypes.js]: >Statement inside of curly braces should be on next line
`brace-style` rule

File: perf/tests/renderComponent-basic.js (hunk 5 above)
Comment (chicoxyzzy) [inline perf/tests/renderComponent-basic.js]: there is `space-before-blocks`. I'll add it

File: test/lib/jasmine-execute.js (hunk 6 above)
Comment (chicoxyzzy) [inline test/lib/jasmine-execute.js]: oops

File: grunt/tasks/eslint.js (hunk 2 above)
Comment (zpao) [inline grunt/tasks/eslint.js]: Can we just use `'.'` then to pick up the root?

File: main.js (hunk 3 above)
Comment (zpao) [inline main.js]: Just add `global Buffer` or whatever the escape is. It's dumb that node puts `Buffer` in the global but such is life. I'd rather the lint escape than write code that looks different from everybody else's.

File: perf/tests/propTypes.js (hunk 4 above)
Comment (zpao) [inline perf/tests/propTypes.js]: Sure, but you created a `func` function and used that earlier to replace the exact same code block. Is there a reason you didn't make the above one an inline function declaration as well?

File: perf/tests/renderComponent-basic.js (hunk 5 above)
Comment (zpao) [inline perf/tests/renderComponent-basic.js]: Please do. For the most part, the rules we have in `src/.eslintrc` are the ones we want everywhere (though I see we don't have that one - perhaps we should. We can do that later since this is already big).

File: grunt/tasks/eslint.js (hunk 2 above)
Comment (chicoxyzzy) [inline grunt/tasks/eslint.js]: no this doesn't work

**Comment by chicoxyzzy** [comment]
> Also, it looks like something got weird with your branching (or GitHub broke). Can you rebase and whatever you need to get that sorted out.

Wow.. it's really weird. I'll check it

**Comment by chicoxyzzy** [comment]
@zpao rebased

File: perf/tests/renderComponent-basic.js (hunk 5 above)
Comment (chicoxyzzy) [inline perf/tests/renderComponent-basic.js]: done

File: grunt/tasks/eslint.js (hunk 2 above)
Comment (hzoo) [inline grunt/tasks/eslint.js]: what about `eslint ./` then?

Comment (chicoxyzzy) [inline grunt/tasks/eslint.js]: @hzoo this doesn't work as well.
BTW `**/*.js` works. But 
```
➜  react git:(remove-jshint) ./node_modules/eslint/bin/eslint.js **/*.js
zsh: argument list too long: ./node_modules/eslint/bin/eslint.js
```
so we need more flexible grunt task or maybe some grunt plugin. But I hope this can be another PR because this diff is too large already.

**Comment by chicoxyzzy** [comment]
@zpao also just noticed you updated `/src/.eslint` in #3133 but it's better to add this rules to `/.eslint` (if this PR will be merged)

**Comment by chicoxyzzy** [comment]
should I merge master now?

**Comment by chicoxyzzy** [comment]
about commit weirdness. I was playing with this [script](https://help.github.com/articles/changing-author-info/) trying to change my name from chico to my real name but I thought I was playing in my test repo and so I broke git history. Then I pushed my broken `react` branch to my remote next day. So I solved that by `-rm -rf` my `.git` directory, reinit git and rebase my branch from `master` to apply my changes on top of it.

**Comment by zpao** [comment]
Alright, lets do it. Thanks for fixing the commit weirdness!

This will definitely be useful and I know it was painstaking work, so we really appreciate it.
